src/Monolog/Handler/TelegramBotHandler.php

```diff
@@ -149,33 +167,105 @@ public function handleBatch(array $records): void
-            'disable_notification' => $this->disableNotification,
-        ]));
+        curl_setopt(
+            $ch,
+            CURLOPT_POSTFIELDS,
+            http_build_query(
+                [
+                    'text' => $message,
+                    'chat_id' => $this->channel,
+                    'parse_mode' => $this->parseMode,
+                    'disable_web_page_preview' => $this->disableWebPagePreview,
+                    'disable_notification' => $this->disableNotification,
+                ]
+            )
+        );
 
-        $result = Curl\Util::execute($ch);
-        $result = json_decode($result, true);
+        return $ch;
+    }
 
-        if ($result['ok'] === false) {
-            throw new RuntimeException('Telegram API error. Description: ' . $result['description']);
+    /**
+     * Handle too long message.
+     *
+     * @return string[]
+     */
+    protected function handleTooLongMessage(string $message): array
+    {
+        // nothing to do with the message if message is inside in the limit
+        if (function_exists('mb_strlen') && mb_strlen($message) <= self::MESSAGE_LENGTH_LIMIT) {
+            return [$message];
+        } elseif (strlen($message) <= self::MESSAGE_LENGTH_LIMIT) {
+            return [$message];
+        }
+
+        // do something with the message if the message is outside of the limit.
+        switch ($this->tooLongMessageHandlingStrategy) {
+            case self::TRIM_TOO_LONG_MESSAGE:
+                // the whole message will be trimmed to the maximum message length.
+                if (function_exists('mb_substr')) {
+                    return [mb_substr($message, 0, self::MESSAGE_LENGTH_LIMIT - 3) . '...'];
+                }
+                return [substr($message, 0, self::MESSAGE_LENGTH_LIMIT - 3) . '...'];
+            case self::SPLIT_TOO_LONG_MESSAGE_INTO_MULTIPLE_MESSAGES:
+                // the message will be chunked into separate messages with the maximum length limit.
+                if (function_exists('mb_str_split')) {
+                    $splittedMessages = mb_str_split($message, self::MESSAGE_LENGTH_LIMIT - 3);
+                    if (!is_array($splittedMessages)) {
+                        throw new RuntimeException('Unable to split too long message');
+                    }
+
+                    $lastChunk = end($splittedMessages);
+                    foreach ($splittedMessages as &$splittedMessage) {
+                        if ($lastChunk === $splittedMessage) {
+                            break;
+                        }
+
+                        $splittedMessage .= '...';
+                    }
+                }
+
+                return str_split($message, self::MESSAGE_LENGTH_LIMIT);
+            // It's most likely cause 'message is too long' exception from Telegram API.
+            case self::DO_NOTHING_WITH_TOO_LONG_MESSAGE:
+            default:
+                return [$message];
```

Comment:
This seems kinda pointless, I assume you did it for BC but if it just causes failures always, I'd rather see some reasonable default strategy in place instead of keeping an old broken strategy. I'd say trimming by default sounds good, and offering splitting as an alternative.